database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java

```diff
@@ -212,11 +244,34 @@ public void write(DataOutput out) throws IOException {
    * @param stmt the {@link PreparedStatement} to write the {@link StructuredRecord} to
    */
   public void write(PreparedStatement stmt) throws SQLException {
+    List<String> updatedKeyList = new ArrayList<>();
+    if (relationTableKey != null) {
+      updatedKeyList = Arrays.asList(relationTableKey.split(","));
+    }
+    modifiableColumnTypes = new ArrayList<>(columnTypes);
     for (int i = 0; i < columnTypes.size(); i++) {
       ColumnType columnType = columnTypes.get(i);
       Schema.Field field = record.getSchema().getField(columnType.getName());
       writeToDB(stmt, field, i);
+      if (fillUpdateParams(updatedKeyList, columnType)) {
+        modifiableColumnTypes.add(columnType);
+      }
     }
+    if (operationName != null && relationTableKey != null) {
```

Comment:
what will be the value of operation here, can it be null in case of insert for other plugins?
And writetoDB method can be called twice here. This logic is not looking good, please have a look

Reply:
operation name cannot be null as by default for other plugins the value being passed is insert. and it will not be called twice because of the modifiableColumnTypes. This will contain the updated size based on the listKeys added to be filled while writing. writeToDB will only be called when modifiableColumnTypes size will be updated. Basically, if it will have field to be updated then the for loop will be executed.